Firestore/core/test/firebase/firestore/remote/grpc_unary_call_test.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <initializer_list>
+#include <memory>
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_unary_call.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+#include "Firestore/core/src/firebase/firestore/util/statusor.h"
+#include "Firestore/core/test/firebase/firestore/util/grpc_stream_tester.h"
+#include "absl/types/optional.h"
+#include "grpcpp/support/byte_buffer.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::AsyncQueue;
+using util::CompletionEndState;
+using util::GrpcStreamTester;
+using util::Status;
+using util::StatusOr;
+using util::CompletionResult::Error;
+using util::CompletionResult::Ok;
+
+class GrpcUnaryCallTest : public testing::Test {
+ public:
+  GrpcUnaryCallTest() : call_{tester_.CreateUnaryCall()} {
+    call_->Start([this](const StatusOr<grpc::ByteBuffer>& result) {
+      status_ = result.status();
+    });
+  }
+
+  ~GrpcUnaryCallTest() {
+    tester_.Shutdown();
+  }
+
+  GrpcUnaryCall& call() {
+    return *call_;
+  }
+  AsyncQueue& worker_queue() {
+    return tester_.worker_queue();
+  }
+
+  void ForceFinish(std::initializer_list<CompletionEndState> results) {
+    tester_.ForceFinish(call_->context(), results);
+  }
+  void KeepPollingGrpcQueue() {
+    tester_.KeepPollingGrpcQueue();
+  }
+
+  const absl::optional<Status>& status() const {
```

Comment:
Rather than implementing getters (like you would for proper classes), for test fixtures, you can just leave the members as public/protected and access them directly. It's encouraged by the gtest team.

eg https://github.com/google/googletest/blob/master/googletest/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests

Reply:
I'd prefer to do a follow-up focusing on testing and address it there. I chiefly tried minimizing boilerplate in test cases (e.g., `worker_queue()` is shorter than `tester_.worker_queue()`).